src/roaring.c

```diff
@@ -129,6 +129,46 @@ void roaring_bitmap_add_many(roaring_bitmap_t *r, size_t n_args,
     }
 }
 
+void roaring_bitmap_contains_multi(roaring_bitmap_t *r, size_t n_args,
+                             const uint32_t *vals, bool *results) {
+    if (n_args == 0 || r == NULL)  {
+        return;
+    }
+    uint32_t prev_container_key = 0;   // previous container key
+    void *container = NULL;     // hold value of last container touched
+    uint8_t typecode = 0;       // typecode of last container touched
+    int32_t container_index = -1;    // current container pos
+    const int container_count= r->high_low_container.size;
+
+    for (size_t i = 0; i < n_args; i++) {
+        if ((container != NULL) && (((prev_container_key^ vals[i]) >> 16) == 0)) {
+            // currently still in the same container
+            results[i] = container_contains(container, vals[i] & 0xFFFF, typecode);
+            continue;
+        }
+
+        if (vals[i] < (prev_container_key & 0xFFFF0000)) {
+            // current value belong to the lower container, reset the index then search from beganing
+            container_index = -1;
+        }
+        // search from the container_index(excluded)
+        container_index = ra_advance_until(&r->high_low_container, vals[i]>>16, container_index);
+        if (container_index == container_count) {
+            results[i] = false;
+            container_index = -1;
+            continue;
+        }
+        if (ra_test_value_at_index(&r->high_low_container, container_index, vals[i])) {
+            container = ra_get_container_at_index(&r->high_low_container, container_index, &typecode);
+            results[i] = container_contains(container, vals[i]&0xFFFF, typecode);
+            prev_container_key = vals[i]; // update previous key, cause we really found a container
+        } else {
+            results[i] = false;
+            container_index = -1;
```

Comment:
Why are you setting it to -1 here?

Reply:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/11133870/168422214-7ee4e874-eaa6-4c29-8884-11352b873081.png">
FYI

Reply:
<img width="886" alt="image" src="https://user-images.githubusercontent.com/11133870/168422233-24de7166-f9aa-4754-89c5-86914f2e9493.png">
FYI